JavaScript/chapter01/p04_palinperm/avc278.js

```diff
@@ -0,0 +1,54 @@
+// Palindrome Permutation: Given a string, write a function to check if it is a permutation of a palindrome. A
+// palindrome is a word or phrase that is the same forwards and backwards. A permutation is a rearrangement of letters.
+// The palindrome does not need to be limited to just dictionary words.
+
+const assert = require("assert");
+
+/**
+ * Checks if the input string can be rearranged into a palindrome
+ * @param  {string} str input string to check against
+ * @return {boolean}    whether the input string can be rearranged into a palindrome
+ *
+ * We don't actually have to rearrange the string to solve this problem; we only need to check its contents.
+ * If the string is of odd  length, then each character in the string has to appear an even number of times, except one.
+ * If the string is of even length, then each character in the string has to appear an even number of times.
+ * We can create a letter frequency map which takes O(N) time to loop through the string, and O(N) space to build.
+ * Looping through the object key-value pairs again is an O(N) operation in the worst case where we loop through it all.
+ *
+ * Runtime: O(N)
+ * Space:   O(N)
+ *
+ */
+const palindromePerm = (str) => {
+  const letterFreq = {};
+  let length = 0;
+  for (const letter of str) {
+    if (letter !== " ") {
```

Comment:
small optional readability suggestion: here (and in L37, L38), the [guard clause pattern](https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html) can be used to reduce nesting and improve readability:

```suggestion
    if (letter === " ") continue;  // or in the next line.
```

Reply:
oops... I actually knew this one :P

Reply:
Completely forgot to update this one. Should be fixed now